index.bs

```diff
@@ -989,47 +1041,111 @@ partial interface HTMLIFrameElement {
+    use the feature, then |report| should be set to False.
+
+    1. Let |policy| be |document|'s [=Document/permissions policy=].
+    1. Let |report-only policy| be |document|'s [=Document/report-only
+       permissions policy=].
+    1. Let |result| be the result of calling <a abstract-op>Check permissions
+       policy</a>, given |policy|,
+       |feature|, |origin|, and |document|'s [=Document/origin=].
+    1. Let |report-only result| be the result of calling <a abstract-op>Check
+       permissions policy</a>, given |report-only policy|, |feature|, |origin|,
+       and |document|'s [=Document/origin=].
+    1. If |report| is True:
+        1. Let |settings| be |document|'s <a>environment settings object</a>.
+        1. If |result| is "<code>Disabled</code>":
+            1. Let |endpoint| be the result of calling <a abstract-op>Get the
+                reporting endpoint for a feature</a> given |feature| and
+                |policy|.
+            1. Call <a abstract-op>Generate report for violation of permissions
+                policy on settings</a> given |feature|, |settings|,
+                "<code>Enforce</code>", and |endpoint|.
+        1. Else, if |report-only result| is "<code>Disabled</code>":
+            1. Let |report-only endpoint| be the result of calling <a
+                abstract-op>Get the reporting endpoint for a feature</a> given
+               |feature| and |report-only policy|.
+            1. Call <a abstract-op>Generate report for violation of permissions
+                policy on settings</a> given |feature|, |settings|,
+                "<code>Report</code>", and |report-only endpoint|.
+    1. Return result
+
+    </div>
+  </section>
+  <section>
+    ## <dfn abstract-op id="get-reporting-endpoint">Get the reporting endpoint for a feature</dfn> ## {#algo-get-reporting-endpoint}
+
+    <div class="algorithm" data-algorithm="get-reporting-endpoint">
+    Given a feature (|feature|) and a permissions policy (|policy|), this
+    algorithm returns a string naming the endpoint to send violation reports to,
+    or null if no such endpoint has been declared in |policy|.
+    1. Let |config| be |policy|'s [=permissions policy/declared policy=]'s
+        <a>reporting configuration</a>.
+    1. If |config|[|feature|] exists, return |config|[|feature|].
+    1. Return null.
+
+    </div>
+  </section>
   <section>
     ## <dfn export abstract-op id="report-permissions-policy-violation">Generate report for violation of permissions policy on settings</dfn> ## {#algo-report-permissions-policy-violation}
 
     <div class="algorithm" data-algorithm="report-permissions-policy-violation">
     Given a [=feature=] (|feature|), an <a>environment settings object</a>
-    (|settings|), and an optional string (|group|), this algorithm generates a
-    <a>report</a> about the <a>violation</a> of the policy for |feature|.
+    (|settings|), a String (|disposition|), and an optional string (|group|),
```

Comment:
The callers of this algorithm seem to pass in as the last argument, an endpoint. But the pre-existing argument here is called `|group|`, which is a confusing name. Maybe there's context here that I'm missing, but I just want to be sure that these are at least corresponding to the same thing here, right? (Just want to double check!)

Reply:
Will change to `endpoint` here; originally the reporting api dealt with named endpoint groups, where you could configure a number of URLs to try in sequence to deliver a report. That has changed now, and other than in a couple of special cases, we should be naming endpoints instead.

Reply:
Addressed below